src/whitenoise/accounts/mod.rs

```diff
@@ -113,10 +111,34 @@ where
                 source: Box::new(e),
             })?;
 
-        // Parse onboarding from JSON
-        let onboarding: OnboardingState =
-            serde_json::from_str(&onboarding_json).map_err(|e| sqlx::Error::ColumnDecode {
-                index: "onboarding".to_string(),
+        let discovery_relays: Vec<RelayUrl> = serde_json::from_str(&discovery_relays)
+            .map(|urls: Vec<String>| {
+                urls.iter()
+                    .filter_map(|url| RelayUrl::parse(url).ok())
+                    .collect::<Vec<_>>()
+            })
+            .map_err(|e| sqlx::Error::ColumnDecode {
+                index: "discovery_relays".to_owned(),
+                source: Box::new(e),
+            })?;
+        let inbox_relays: Vec<RelayUrl> = serde_json::from_str(&inbox_relays)
+            .map(|urls: Vec<String>| {
+                urls.iter()
+                    .filter_map(|url| RelayUrl::parse(url).ok())
+                    .collect::<Vec<_>>()
+            })
+            .map_err(|e| sqlx::Error::ColumnDecode {
+                index: "inbox_relays".to_owned(),
+                source: Box::new(e),
+            })?;
+        let key_package_relays: Vec<RelayUrl> = serde_json::from_str(&key_package_relays)
+            .map(|urls: Vec<String>| {
+                urls.iter()
+                    .filter_map(|url| RelayUrl::parse(url).ok())
+                    .collect::<Vec<_>>()
+            })
```

Comment:
We could probably extract the parsing logic to a function to avoid a bit of repetition